includes/core-changes/cryptography/5.0/instantiating-default-implementations-of-cryptographic-abstractions-not-supported.md

```diff
@@ -0,0 +1,130 @@
+CryptoConfig.AddAlgorithm(typeof(SHA256CryptoServiceProvider), typeof(HashAlgorithm).FullName);
+alg = HashAlgorithm.Create();
+// Prints 'System.Security.Cryptography.SHA256CryptoServiceProvider'.
+Console.WriteLine(alg.GetType());
+```
+
+You can also use [machine-wide configuration](../../../../docs/framework/configure-apps/map-algorithm-names-to-cryptography-classes.md) to change the default algorithm without needing to call into `CryptoConfig` programmatically.
+
+In .NET Core 2.0 - 3.1, abstract cryptographic primitive factories such as <xref:System.Security.Cryptography.HashAlgorithm.Create?displayProperty=nameWithType> always throw a <xref:System.PlatformNotSupportedException>.
+
+```csharp
+// Throws PlatformNotSupportedException on .NET Core.
+HashAlgorithm alg = HashAlgorithm.Create();
+```
+
+In .NET 5.0 and later versions, abstract cryptographic primitive factories such as <xref:System.Security.Cryptography.HashAlgorithm.Create?displayProperty=nameWithType> are marked obsolete, and produce a compile-time warning with ID `SYSLIB0007`. At run time, these methods continue to throw a <xref:System.PlatformNotSupportedException>.
+
+```csharp
+// Throws PlatformNotSupportedException.
+// Also produces compile-time warning SYSLIB0007 on .NET 5+.
+HashAlgorithm alg = HashAlgorithm.Create();
+```
+
+This is a compile-time only change. There is no run-time change from previous versions of .NET Core.
+
+> [!NOTE]
+>
+> - Only the parameterless overloads of the `Create()` methods are obsolete. Parameterized overloads are not obsolete and still function as expected.
+>
+>   ```csharp
+>   // Call Create(string), providing an explicit algorithm family name.
+>   // Works in .NET Framework, .NET Core, and .NET 5.0+.
+>   HashAlgorithm hashAlg = HashAlgorithm.Create("SHA256");
+>   ```
+>
+> - Parameterless overloads of *specific* algorithm families (not abstractions) are not obsolete, and will continue to function as expected.
+>
+>   ```csharp
+>   // Call a specific algorithm family's parameterless Create() ctor.
+>   // Works in .NET Framework, .NET Core, and .NET 5.0+.
+>   Aes aesAlg = Aes.Create();
+>   ```
+
+#### Reason for change
+
+The cryptographic configuration system present in .NET Framework is no longer present in .NET Core and .NET 5.0+, since it does not allow for proper cryptographic agility. .NET's backward-compatibility requirements also prohibit the framework from updating certain cryptographic APIs to keep up with advances in cryptography. For example, the <xref:System.Security.Cryptography.HashAlgorithm.Create?displayProperty=nameWithType> method was introduced in .NET Framework 1.0, when the SHA-1 hash algorithm was state-of-the-art. Twenty years have passed, and now SHA-1 is considered broken, but we cannot change <xref:System.Security.Cryptography.HashAlgorithm.Create?displayProperty=nameWithType> to return a different algorithm. Doing so would introduce an unacceptable breaking change in consuming applications.
```

Comment:
```suggestion
The cryptographic configuration system present in .NET Framework is no longer present in .NET Core and .NET 5.0+, since that legacy system doesn't allow for proper cryptographic agility. .NET's backward-compatibility requirements also prohibit the framework from updating certain cryptographic APIs to keep up with advances in cryptography. For example, the <xref:System.Security.Cryptography.HashAlgorithm.Create?displayProperty=nameWithType> method was introduced in .NET Framework 1.0, when the SHA-1 hash algorithm was state-of-the-art. Twenty years have passed, and now SHA-1 is considered broken, but we cannot change <xref:System.Security.Cryptography.HashAlgorithm.Create?displayProperty=nameWithType> to return a different algorithm. Doing so would introduce an unacceptable breaking change in consuming applications.
```